vllm_spyre/v1/worker/spyre_input_batch.py

```diff
@@ -312,21 +330,26 @@ def soft_remove_request(self, req_id: str):
         This method however, does not change the self.req_id_to_index which is
         needed to keep the map of finished padded requests.
         
-        TODO: in the future we shall remove/change this method to actual clean 
-        up all the data 
+        For the continuous batching, the removed request indices can be 
+        overwritten by new requests
         '''
         if req_id not in self.req_id_to_index:
             return
-        req_index = self.req_id_to_index[req_id]
+        # req_index = self.req_id_to_index[req_id]
+        req_index = self.req_id_to_index.pop(req_id)
+
+        if req_index is None:
+            return None
```

Comment:
Since we're not really returning anything
```suggestion
            return
```